tests/test_app_events.py

```diff
@@ -1,66 +1,46 @@
 """Test events occurring on app startup or shutdown."""
 
-import os
 import warnings
 
-import httpx
 import pytest
 
 from app.api import utility as util
+from app.main import settings
 
 
 @pytest.mark.filterwarnings("ignore:.*NB_API_ALLOWED_ORIGINS")
 def test_start_app_without_environment_vars_fails(
-    test_app, monkeypatch, disable_auth
+    test_app, disable_auth, monkeypatch
 ):
     """Given non-existing username and password environment variables, raises an informative RuntimeError."""
-    monkeypatch.delenv(util.GRAPH_USERNAME.name, raising=False)
-    monkeypatch.delenv(util.GRAPH_PASSWORD.name, raising=False)
+    monkeypatch.setattr(settings, "graph_username", None)
+    monkeypatch.setattr(settings, "graph_password", None)
 
```

Comment:
why do we have to monkeypatch this now? I expected that pytest-env allows us to just "mock" the env variable in from the perspective of the TestClient?

Reply:
ah, I see. pytest-env only allows us to set one group of ENV variables for **all** test we're running. Hhm. How useful do you think that is then?

Reply:
> pytest-env only allows us to set one group of ENV variables for all test we're running

yes, at least from my understanding!

> How useful do you think that is then?

I think it's still useful because it means we can actually test how environment variables are read in by the app and stored in corresponding internal variables, which is different than testing the app's behaviour assuming a specific value for an internal variable (i.e., once an env var has already been read in). This is important because env vars are always read in as strings, but we have several variables we expect to be **non-strings** when referenced internally within the app (e.g., they should be stored as bools or ints). This is hard to test without pytest-env.

Reply:
(just realized I drafted but never posted this comment)
this test also serves as a good precursor for:

- https://github.com/neurobagel/api/issues/411

in that we've seen there are cases where whether the default value has been applied is not obvious (if a variable has been unset vs. set to ""). We could confirm the actual behaviour in a test such as this one!